### ISSUE TITLE
How to reduce the encoding times when FreeSWITCH is recording a video

### ISSUE THREAD

**Issue by visitewen** [signalwire/freeswitch#2608]
Is it possible to reduce the encoding times when FreeSWITCH is recording a video,as i know,when FS recording video, will re-codec and transmit,when aleg and bleg have same video codec,I think no need re-codec to transmit,only re-codec for recording.

**Comment by seven1240** [comment]
yes, with more if-else. I think I had a patch that could record the encoded stream to mp4 w/o decoding at all.

**Comment by visitewen** [comment]
can you share more detail steps or how to get the patch,really  appreciate！

**Comment by seven1240** [comment]
It was a long time, not many people using that, I think it might be in the https://signalwire.com/products/freeswitch-enterprise .

**Comment by visitewen** [comment]
commercial support？

**Comment by tomek-o** [comment]
> I think I had a patch that could record the encoded stream to mp4 w/o decoding at all.

@seven1240 I believe you mean mod_mp4v2 and it is still there.
mod_fsv is another options.
Some tweaking to FS core might be needed depending on recording context, but not much.

**Comment by visitewen** [comment]
@tomek-o  I want to transmit the original video data（not re-coding） to the terminal while recording. Can you give me some detailed suggestions? Many thanks.

**Comment by tomek-o** [comment]
@visitewen I was doing it at work, so I can't. Start with existing modules, the number of issues to solve would depend on actual endpoints, codecs in use and call scenarios.

**Comment by visitewen** [comment]
Ok,Thank you.
